backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/migration/CassandraMigrationService.java

```diff
@@ -51,12 +49,12 @@ public CassandraMigrationService(CassandraSchemaVersionDAO schemaVersionDAO, Map
         this.allMigrationClazz = allMigrationClazz;
     }
 
-    public CassandraVersionResponse getCurrentVersion() {
-        return new CassandraVersionResponse(schemaVersionDAO.getCurrentSchemaVersion().join());
+    public Optional<Integer> getCurrentVersion() {
```

Comment:
no more strong typing on version?

Reply:
CassandraVersionResponse is a webadmin DTO. In order to extract this out, we need to not rely on it anymore.

That being said, I propose introducing a **CassandraSchemaVersion** throughout that codebase to replace that Integer (refactoring).

IE, the problem is the DAO not returning strongly typed objects.